### PR TITLE
Cookie policies for iframe usage

### LINKED ISSUE #265: [Bug] Translations reset on route change in production
## Description

Translations reset themselves to english on any navigation in the production environment. Is likely related to the cookie policy and using an iFrame.

## Reproduction

1. Go to es.wp.org/openverse
2. Click on any internal url in the Openverse iFrame.
3. Notice that the UI has reverted to English.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### LINKED ISSUE #266: [Bug] Incorrect cookie policies for iFrame use
## Description

Cookies set by Openverse, for our A/B tests and for i18n support, set incorrect cookie options. They need to use:

- SameSite: None
- Secure: true

Which allows for sharing of third-party cookies on HTTPS-enabled sites.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #266 by @zackkrida
Fixes #265 by @zackkrida

## Description

This PR implements the cookie policies necessary for usage with an iFrame. This should fix the production translation issue as well.

## Tests

1. This requires testing localhost securely, so try the following steps:
2. Build Openverse locally with `npm run build`
3. Run `brew install mkcert nss` to easily manage local certificates
4. Run `mkcert -install` so your machine automatically trusts created certificates
4. Run `mkcert localhost` in the root of the project
5. Add the following to `nuxt.config.js`:
```js
// at the top
import fs from 'fs'
import path from 'path'

// in the "server" block:
https: {
  key: fs.readFileSync(path.resolve(__dirname, 'localhost-key.pem')),
  cert: fs.readFileSync(path.resolve(__dirname, 'localhost.pem'))
}
```
6. Run `npm run start` to start Openverse securely at `https://localhost:8443/`
7. Go to es.wordpress.org/openverse and use devtools to replace the iFrame URL with `https://localhost:8443/`
8. Confirm that translations are persistient now! Optionally verify the cookie is set properly.

## Checklist 
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/utils/session-id.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { v4 as uuidv4 } from 'uuid'
 import Cookie from 'js-cookie'
 
 const COOKIE_EXPIRY_DAYS = 7
+const COOKIE_PREFIX = 'openverse_'
```
Comment (zackkrida) [inline src/utils/session-id.js:5]: This is a small naming improvement but also meant to clear any cached cookies with the old cookie policies.

File: src/utils/session-id.js [hunk 2]
```diff
@@ -16,7 +17,11 @@ const generateSessionId = () => uuidv4()
  */
 const saveSessionIdInCookie = (cookieName, sessionId, hasExpirationDate) => {
   if (hasExpirationDate) {
-    Cookie.set(cookieName, sessionId, { expires: COOKIE_EXPIRY_DAYS })
+    Cookie.set(COOKIE_PREFIX + cookieName, sessionId, {
```
Comment (dhruvkb) [inline src/utils/session-id.js:20]: Just a personal preference if I may point out one:
```suggestion
    Cookie.set(`${COOKIE_PREFIX}${cookieName}`, sessionId, {
```

**Review by dhruvkb — APPROVED** [review]
LGTM!

**Review by krysal — APPROVED** [review]
It works perfectly. Thanks for the detailed instructions 🌟

File: src/utils/session-id.js [hunk 3]
```diff
@@ -28,7 +33,7 @@ const saveSessionIdInCookie = (cookieName, sessionId, hasExpirationDate) => {
  * @param {boolean} [hasExpirationDate=false] Whether the cookie should expire when user closes the browser
  */
 const SessionId = (cookieName, hasExpirationDate = false) => {
-  let sessionId = Cookie.get(cookieName)
+  let sessionId = Cookie.get(COOKIE_PREFIX + cookieName)
```
Comment (zackkrida) [inline src/utils/session-id.js:36]:
```suggestion
  let sessionId = Cookie.get(`${COOKIE_PREFIX}${cookieName}`)
```

**Pull request merged**